src/metabase/api/table.clj

```diff
@@ -79,6 +80,14 @@
         (sync-database/sync-table! updated-table))
       updated-table)))
 
+(defn- format-fields-for-response [resp]
+  (update resp :fields
+          (fn [fields]
+            (map (fn [{:keys [values] :as field}]
+                   (if (seq values)
+                     (update field :values fv/field-values->pairs)
+                     field))
+                 fields))))
```

Comment:
I think a `for` loop would be a little more readable here, e.g.

```clojure
(defn- format-fields-for-response [resp]
  (update resp :fields 
          (fn [fields]
            (for [{:keys [values] :as field} fields]
              (if (seq values)
                (update field :values fv/field-values->pairs)
                field)))))
```

Generally I think it's better to use higher-level constructs like `for` over `map` where sensible